### PR TITLE
Add delete job offer  niousha mahjoub

### PR DESCRIPTION
**Pull request by nioushamahjoub**
Add delete job offer

### CODE REVIEW

**Review by MortezaGitPro — APPROVED** [review]

File: src/main/java/com/soen6011/SOEN6011/Controller/Soen6011Controller.java [hunk 1]
```diff
@@ -55,5 +56,14 @@ public String myJobOffers(Model model)
 
         return "my-job-offers";
     }
+
+    @GetMapping("/deleteJobOffer/{id}")
+    public String deleteJobOffer(@PathVariable("id")int id) {
+        jobOfferRepository.deleteById(id);
```
Comment (MajorTreble) [inline src/main/java/com/soen6011/SOEN6011/Controller/Soen6011Controller.java:62]: What is the id is not found?

File: src/main/resources/templates/my-job-offers.html [hunk 2]
```diff
@@ -48,11 +48,34 @@ <h5 th:text="${jobOffer.title}"></h5>
           <p><strong>Low Salary:</strong> <span th:text="${jobOffer.lowsalery}"></span></p>
           <p><strong>High Salary:</strong> <span th:text="${jobOffer.highsalery}"></span></p>
         </div>
+        <div class="card-footer">
+          <a class="btn btn-danger btn-sm" th:href="'/deleteJobOffer/'+${jobOffer.id}">Delete</a>
```
Comment (MajorTreble) [inline src/main/resources/templates/my-job-offers.html:52]: Does this make sense as a GET?
Ask for confirmation before deleting.

File: src/main/resources/templates/my-job-offers.html [hunk 3]
```diff
@@ -48,11 +48,34 @@ <h5 th:text="${jobOffer.title}"></h5>
...
+    if (confirmation) {
+      fetch(`/deleteJobOffer?id=${id}`, {
+        method: 'GET'
+      })
+      .then(response => {
+        if (response.ok) {
+          // Reload the page to see the updated job offers after deletion.
+          location.reload();
+        } else {
+          console.error('Failed to delete job offer:', response.status);
```
Comment (MajorTreble) [inline src/main/resources/templates/my-job-offers.html:72]: This is not a failure, they canceled.

File: src/main/resources/templates/my-job-offers.html [hunk 4]
```diff
@@ -48,11 +48,34 @@ <h5 th:text="${jobOffer.title}"></h5>
...
+        }
+      })
+      .catch(error => console.error('Error deleting job offer:', error));
```
Comment (MajorTreble) [inline src/main/resources/templates/my-job-offers.html:75]: Does console go back to the user?

**Comment by MajorTreble** [comment]
No unit tests
No javadoc

File: src/main/resources/templates/my-job-offers.html (hunk 4 above)
Comment (nioushamahjoub) [reply src/main/resources/templates/my-job-offers.html:75]: they are just for test and see what happens. I will remove them.

**Pull request merged**